### ISSUE TITLE
High-dim reach should also return options

### ISSUE THREAD

**Issue by mforets** [JuliaReach/Reachability.jl#11]
Instead of:

```julia
julia> s = solve(ContinuousSystem(A, X0), :N => 10, :δ=> 0.1, :blocks=>[2, 5], :apply_projection=>false);

julia> s.options
ERROR: type Array has no field options
```

it should return the tuple `(polygons, options)` (or another name instead of `polygons`). one reason is that we need the options structure to call the `project` function.

**Comment by schillic** [comment]
The problem is that without projection we [return the raw result](https://github.com/JuliaReach/Reachability.jl/blob/f26090f69462e1713609c08a7cc6b9d68d7bf66b/src/solve.jl#L114).

We should probably parameterize the `Rsets2DSolution` type with the type of the `LazySet` that it uses in its vector. Then we can just wrap the result as we do a few lines above for the projection result.

**Comment by mforets** [comment]
fixed in PR #14
